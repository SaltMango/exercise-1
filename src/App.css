.time {
  font-family: Roboto-Bold;
  font-size: 84px;
  color: #37474f;
  text-align: center;
  margin: 0;
}
.container{
  height: 97vh;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0;
}
