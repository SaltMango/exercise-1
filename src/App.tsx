import React, { useEffect, useState } from "react";
import "./App.css";

const app:React.FC = () => {
  const [date, setDate] = useState<number>(Date.now());

  useEffect(() => {
    let timer: NodeJS.Timeout = setInterval(() => setDate(Date.now()), 1000);
    return () => {
      clearInterval(timer);
    };
  });

  let time: string = new Date(date).toLocaleString("en-US", {
    hour12: true,
    hour: "2-digit",
    minute: "numeric",
    second: "numeric",
  });
  return (
    <div className="container">
      <h1 className="time">{time}</h1>
    </div>
  );
};

export default app;
