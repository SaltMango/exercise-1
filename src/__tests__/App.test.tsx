import App from "../App";
import React from "react";
import TestRenderer from "react-test-renderer";

Date.now = jest.fn(() => new Date("2020-05-25T12:20:30Z").getTime());
test("time renders correctly", () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});