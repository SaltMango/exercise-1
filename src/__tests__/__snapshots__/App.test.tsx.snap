// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time renders correctly 1`] = `
<div
  className="container"
>
  <h1
    className="time"
  >
    05:50:30 PM
  </h1>
</div>
`;
